### PR TITLE
[#173017733] start activation checks and request creation

### PR DESCRIPTION
**Pull request by balanza**
Implementing the http controller to start the bonus activation procedure. I leave it draft as I need to write some uni test for it.

### What's included
* check if the user has a pending eligibility check request
* check if the user has a pending bonus activation request
* check if the user has a valid DSU
* creates the bonus activation record on the db

### Changes
* added 410 Gone response status to the spec https://github.com/pagopa/io-functions-bonus/blob/93ada8d6607add541245e68a7905ccc326614656/openapi/index.yaml#L102
* created global builder for orchestrators' identifiers https://github.com/pagopa/io-functions-bonus/blob/f59c7a7b121a4de8c2a3285b0ea20eab2fcb59e8/utils/orchestrators.ts#L1
and implemented [here](https://github.com/pagopa/io-functions-bonus/pull/31/files#diff-f10cefa5a25799628090ea89be782a87) and [here](https://github.com/pagopa/io-functions-bonus/pull/31/files#diff-1879ec4a1ee882fb1b4a9f0dd6c17f7f)

### CODE REVIEW

File: StartBonusActivation/handler.ts [hunk 1]
```diff
@@ -1,50 +1,295 @@
...
+    const bonusActivation: NewBonusActivation = {
+      applicantFiscalCode: fiscalCode,
+      code: bonusCode,
+      dsuRequest: dsu,
+      id: bonusCode,
+      kind: "INewBonusActivation",
+      status: BonusActivationStatusEnum.PROCESSING,
+      updatedAt: new Date()
+    };
+    return bonusActivationModel.create(bonusActivation, fiscalCode);
```
[the remark this answers is not in the document]
Comment (balanza) [reply StartBonusActivation/handler.ts:215]: we leave this for another PR

File: StartBonusActivation/handler.ts [hunk 2]
```diff
@@ -1,51 +1,279 @@
...
+          | IResponseErrorGone
+          | IResponseErrorInternal,
+          Dsu
+        >
+      >(fromEither(left(ResponseErrorInternal(`Cannot find DSU`))), doc =>
+        // the found document is not in eligible status
+        !EligibilityCheckSuccessEligible.is(doc)
+          ? fromEither(left(ResponseErrorForbiddenNotAuthorized))
+          : // the check is expired
+          doc.validBefore > new Date()
```
Comment (gunzip) [inline StartBonusActivation/handler.ts:178]: consider using date-fns https://github.com/pagopa/io-functions-bonus/blob/f32808205244d297c895f0922236a1fc4956e75f/GetEligibilityCheck/handler.ts#L102

File: StartBonusActivation/handler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gunzip) [reply StartBonusActivation/handler.ts:215]: can you add a TODO for every missing part of this activity ?

File: package.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
     "predev": "npm-run-all generate build",
     "dev": "npm-run-all --parallel start watch",
     "pretest": "npm-run-all generate:*",
-    "test": "jest",
+    "test": "yarn test:only",
```
Comment (gunzip) [inline package.json:17]: why this change?

File: utils/responses.ts [hunk 4]
```diff
@@ -11,7 +11,8 @@ export interface IResponseErrorGone extends IResponse<"IResponseErrorGone"> {
  */
 export function ResponseErrorGone(detail: string): IResponseErrorGone {
```
Comment (gunzip) [inline utils/responses.ts:12]: we can remove this file and use https://github.com/pagopa/io-ts-commons/blob/master/src/responses.ts#L550 
(I've added it today)

File: package.json (hunk 3 above)
Comment (balanza) [reply package.json:17]: `yarn test` perfoms code generation before actually running test. This is very unpleasant when working with a short dev-test-refactor cycle. So I introduced a `yarn test:only` script that basically doesn't trigger `pretest`.

File: StartBonusActivation/handler.ts [hunk 5]
```diff
@@ -1,51 +1,283 @@
...
     const client = df.getClient(context);
-    const status = await client.getStatus(
-      makeStartBonusActivationOrchestratorId(fiscalCode)
-    );
-    if (status.runtimeStatus === df.OrchestrationRuntimeStatus.Running) {
-      return ResponseSuccessAccepted();
-    }
-    return ResponseErrorConflict("Implementation missing");
+
+    // TODO: lock for familiuid
```
Comment (gunzip) [inline StartBonusActivation/handler.ts:239]: sorry I meant: these TODOs sould be put wherever the implementation goes (not as a checklist before everything else : )

File: utils/responses.ts (hunk 4 above)
Comment (balanza) [reply utils/responses.ts]: done

File: StartBonusActivation/handler.ts (hunk 5 above)
Comment (balanza) [reply StartBonusActivation/handler.ts]: done
